src/wallet/wallet.h

```diff
@@ -722,6 +725,14 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
 
     std::unique_ptr<CWalletDBWrapper> dbw;
 
+    /**
+     * The following is used to keep track of how far behind the wallet is
+     * from the chain sync, and to allow clients to block on us being caught up
+     *
+     * Protected by cs_main
```

Comment:
I don't understand why this is protected by cs_main. Can you explain?

Reply:
The code in BlockUntilSyncedToCurrentChain accesses this in a place that's going to need cs_main anyway (for chainActive.Tip()), could add a cs_wallet and call it protected by cs_wallet, but that just means an extra lock in BlockUntilSyncedToCurrentChain.

Reply:
I added comments to explain this.